### PR TITLE
select2_blog

### PR DESCRIPTION
**Pull request by bmrsny**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/wombatsecurity/wombatsecurity.github.io/pull/41" target="_blank">ReviewNinja

File: _posts/2015-12-2-why-select1-when-you-can-select2.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+categories: development
+comments: true
+author: Brandon Mrsny
+author_github: bmrsny
+banner_image: choices.jpg
+---
+
+### Choices Choices Choices
+
+Creating unique and painless layouts for users becomes integral to your applications usage with it's Customers. Leveraging front end tools can help achieve a more pleasant experience not only for users, but the developer as well. Just to be clear, this is not going to be a post about UX/Design Patterns. Mainly a focus on a jQuery library, [Select2](http://select2.github.io/select2/).
```
Comment (laurawhalin) [inline _posts/2015-12-2-why-select1-when-you-can-select2.md]: edit to: "integral to your application's usage with its customers."

File: _posts/2015-12-2-why-select1-when-you-can-select2.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+---
+
+### Choices Choices Choices
+
+Creating unique and painless layouts for users becomes integral to your applications usage with it's Customers. Leveraging front end tools can help achieve a more pleasant experience not only for users, but the developer as well. Just to be clear, this is not going to be a post about UX/Design Patterns. Mainly a focus on a jQuery library, [Select2](http://select2.github.io/select2/).  
+
+You most likely have scrolled through endless drop down options in search of the proper value. Sometimes glancing over the very selection you want. I've encountered some drop downs that extended beyond the window of my browser. It looks painful, and feels equally painful.  
+<img style="padding-left: 20px; width: 400px" src="{{ site.url }}/assets/images/sad_drop_down.jpg">
+
+In modern applications, drop down menus hold database values for Users to select from. Loading them up on the server and iterating and laying them out on the front end. Within the ThreatSim platform, we utilize the Rails Framework. Here is what using a basic drop down in Rails looks like.
```
Comment (laurawhalin) [inline _posts/2015-12-2-why-select1-when-you-can-select2.md]: I'd ditch the first period and put a comma instead, so the second sentence isn't a fragment.

File: _posts/2015-12-2-why-select1-when-you-can-select2.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+<div id="mySelect">
+  <select id="25">
+      <option value="AL"> Alaska</option>
+     <option value="CO"> Colorado</option>
+   </select>
+</div>
+
+__Why Select2?__
+
+Select2 allows you as the developer to transform your select box into something more powerful. Giving your User the ability to search for values within the drop down.
```
Comment (laurawhalin) [inline _posts/2015-12-2-why-select1-when-you-can-select2.md]: Same here, change the first period to a comma.

File: _posts/2015-12-2-why-select1-when-you-can-select2.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+{% highlight javascript %}
+  $("#mySelect").select2();
+{% endhighlight  %}
+
+That is all it takes to get started!
+
+__Options__
+
+ But that is not all. Select2 comes with a multitude of options to add to your custom select2 box. You have options for placeholder values, minimum character Input for searches, templating, and more. One of my favorites is the multiple select feature.
```
Comment (laurawhalin) [inline _posts/2015-12-2-why-select1-when-you-can-select2.md]: Input does not need to be capitalized.

File: _posts/2015-12-2-why-select1-when-you-can-select2.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+ To work with options you just need to add these properties to your select2 initialization. 
+
+{% highlight javascript %}
+  $("#mySelect").select2({
+    multiple: true // select multiple options from the drop down
+  });
+{% endhighlight  %}
+
+What happens when your drop down starts becoming too cluttered, or you need to update your select box with new values based on User interactions. Well, Select2 comes with the ability to load remote data with Ajax.
```
Comment (laurawhalin) [inline _posts/2015-12-2-why-select1-when-you-can-select2.md]: Change the first period to a ?

File: _posts/2015-12-2-why-select1-when-you-can-select2.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        results: results
+      };
+    }
+  }
+});
+{% endhighlight %}
+The one trick is returning an id and text for the select box.
+That is about it! A breeze to use and develop with! Check out the Docs for more info on using select2 and all the options! [Select2](http://select2.github.io/select2/).
+
+Now go Select all the things!
```
Comment (laurawhalin) [inline _posts/2015-12-2-why-select1-when-you-can-select2.md]: Select doesn't need to be capitalized.

**Pull request merged**